cdap-app-fabric/src/main/java/io/cdap/cdap/support/task/SupportBundleRuntimeInfoTask.java

```diff
@@ -0,0 +1,141 @@
+  private final ProgramType programType;
+  private final ProgramId programName;
+  private final RemoteMetricsSystemClient remoteMetricsSystemClient;
+  private final File appPath;
+  private final List<RunRecord> runRecordList;
+  private final ApplicationDetail applicationDetail;
+
+  @Inject
+  public SupportBundleRuntimeInfoTask(File appPath, NamespaceId namespaceId, ApplicationId appId,
+                                      ProgramType programType, ProgramId programName,
+                                      RemoteMetricsSystemClient remoteMetricsSystemClient,
+                                      List<RunRecord> runRecordList, ApplicationDetail applicationDetail) {
+    this.namespaceId = namespaceId;
+    this.appId = appId;
+    this.programType = programType;
+    this.programName = programName;
+    this.appPath = appPath;
+    this.remoteMetricsSystemClient = remoteMetricsSystemClient;
+    this.runRecordList = runRecordList;
+    this.applicationDetail = applicationDetail;
+  }
+
+  @Override
+  public void collect() throws IOException, NotFoundException, JSONException {
+    for (RunRecord runRecord : runRecordList) {
+      String runId = runRecord.getPid();
+      try (FileWriter file = new FileWriter(new File(appPath, runId + ".json"))) {
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.addProperty("status", runRecord.getStatus().toString());
+        jsonObject.addProperty("start", runRecord.getStartTs());
+        jsonObject.addProperty("end", runRecord.getStopTs());
+        jsonObject.addProperty("profileName", runRecord.getProfileId().getProfile());
+        jsonObject.addProperty("runtimeArgs", runRecord.getProperties().get("runtimeArgs"));
+        JsonObject metrics =
+          queryMetrics(runId, applicationDetail.getConfiguration(), runRecord != null ? runRecord.getStartTs() : 0,
+                       runRecord != null && runRecord.getStopTs() != null ? runRecord.getStopTs() : DateTime.now()
+                         .getMillis());
+        jsonObject.add("metrics", metrics);
+        gson.toJson(jsonObject, file);
+      }
+    }
+  }
+
+  public JsonObject queryMetrics(String runId, String configuration, long startTs, long stopTs) {
+    JsonObject metrics = new JsonObject();
+    try {
+      JSONObject appConf =
+        configuration != null && configuration.length() > 0 ? new JSONObject(configuration) : new JSONObject();
+      List<String> metricsList = new ArrayList<>();
+      JSONArray stages = appConf.has("stages") ? appConf.getJSONArray("stages") : new JSONArray();
+      for (int i = 0; i < stages.length(); i++) {
+        JSONObject stageName = stages.getJSONObject(i);
+        metricsList.add(String.format("user.%s.records.out", stageName.getString("name")));
+        metricsList.add(String.format("user.%s.records.in", stageName.getString("name")));
+        metricsList.add(String.format("user.%s.process.time.avg", stageName.getString("name")));
```

Comment:
At the very least this should be configurable or we should just grab all metrics